### PR TITLE
Fixes typing issues in pnpm

### PR DESCRIPTION
**Pull request by Rotorsoft**
See ticket

## Link to Issue
Closes: #7768

## Description of Changes
- Simplifies typings and models
- Fixes tests

### CODE REVIEW

File: packages/commonwealth/server/controllers/server_communities_methods/delete_community.ts [hunk 1]
```diff
@@ -122,9 +117,7 @@ export async function __deleteCommunity(
             },
           );
 
-          const models: Array<
-            ModelStatic<ModelInstance<{ community_id?: string }>>
-          > = [
+          const models: ModelStatic<Model>[] = [
```
Comment (timolegros) [inline packages/commonwealth/server/controllers/server_communities_methods/delete_community.ts:120]: This isn't the correct type -> the type for this should be any model that has `community_id` defined which this type does not enforce.

File: packages/commonwealth/server/controllers/server_communities_methods/update_community_id.ts [hunk 2]
```diff
@@ -70,26 +66,25 @@ export async function __updateCommunityId(
...
-        this.models.StarredCommunity,
-        this.models.Vote,
-        this.models.Webhook,
-        this.models.CommunityContract,
-        this.models.CommunityStake,
-        this.models.DiscordBotConfig,
-        this.models.Group,
-        this.models.Subscription,
-      ];
+    const models: ModelStatic<Model>[] = [
```
Comment (timolegros) [inline packages/commonwealth/server/controllers/server_communities_methods/update_community_id.ts:69]: Same Here

**Review by timolegros — CHANGES_REQUESTED** [review]

File: packages/commonwealth/server/controllers/server_communities_methods/delete_community.ts (hunk 1 above)
Comment (Rotorsoft) [reply packages/commonwealth/server/controllers/server_communities_methods/delete_community.ts]: fixed

File: packages/commonwealth/server/controllers/server_communities_methods/update_community_id.ts (hunk 2 above)
Comment (Rotorsoft) [reply packages/commonwealth/server/controllers/server_communities_methods/update_community_id.ts]: fixed

**Review by timolegros — APPROVED** [review]

**Review by kurtisassad — APPROVED** [review]

**Pull request merged**